### PR TITLE
Inject non-superfluous quotes when rewriting yaml

### PR DESCRIPTION
**Pull request by ewiseblatt**
@jtk54 
Write back quotes into YAML files when they are required.

### CODE REVIEW

**Comment by ewiseblatt** [comment]
@jtk54, added the other change I mentioned. ready for you to review.

File: pylib/spinnaker/transform_old_config.py [hunk 1]
```diff
@@ -20,6 +20,13 @@
 
 
 class Processor(object):
+  def __collect_environ(self, content):
+    result = {}
+    for match in re.finditer('^([^\s]+?)=([^\n]*)(?:\#.+)?', content,
```
Comment (jtk54) [inline pylib/spinnaker/transform_old_config.py]: `[^\s]` should go to `[\S]` in the first group. Are you trying to cut off `#` comments with the second two groups? If so, I would change `[^\n]` to `[^\n\#]` to not match comments before the non-capturing group.

File: pylib/spinnaker/transform_old_config.py [hunk 2]
```diff
@@ -20,6 +20,13 @@
...
+                             re.MULTILINE):
+      result[match.group(1)] = match.group(2)
```
Comment (jtk54) [inline pylib/spinnaker/transform_old_config.py]: You could do some pythonic dictionary construction like `result = {match.group(1): match.group(2) for match in re.finditer(...)}`. If there are no matches this dictionary comprehension would still return `{}`.

**Comment by jtk54** [comment]
LGTM

**Pull request merged**